components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/certificatemgt/CertificateManagerImpl.java

```diff
@@ -0,0 +1,174 @@
+                        " Hence DB entry is removed.");
+            } else {
+                log.info("Certificate is successfully removed from the Publisher Trust Store with Alias '"
+                        + alias + "'");
+            }
+            return responseCode;
+        } else {
+            log.error("Failed to remove certificate from the data base. No certificate changes will be affected.");
+            return ResponseCode.INTERNAL_SERVER_ERROR;
+        }
+    }
+
+    @Override
+    public boolean addCertificateToGateways(String certificate, String alias) {
+        boolean result;
+        ResponseCode responseCode = certificateMgtUtils.addCertificateToTrustStore(certificate, alias);
+        if (responseCode == ResponseCode.ALIAS_EXISTS_IN_TRUST_STORE) {
+            log.info("The Alias '" + alias + "' exists in the Gateway Trust Store.");
+            result = true;
+        } else {
+            result = responseCode != ResponseCode.INTERNAL_SERVER_ERROR;
+        }
+        result = result && touchConfigFile();
+        if (result) {
+            log.info("The certificate with Alias '" + alias + "' is successfully added to the Gateway " +
+                    "Trust Store.");
+        } else {
+            log.error("Error adding the certificate with Alias '" + alias + "' to the Gateway Trust Store");
+        }
+        return result;
+    }
+
+    @Override
+    public boolean deleteCertificateFromGateways(String alias) {
+        ResponseCode responseCode = certificateMgtUtils.removeCertificateFromTrustStore(alias);
+        if (responseCode != ResponseCode.INTERNAL_SERVER_ERROR) {
+            log.info("The certificate with Alias '" + alias + "' is successfully removed from the Gateway " +
+                    "Trust Store.");
+        } else {
+            log.error("Error removing the certificate with Alias '" + alias + "' from the Gateway " +
+                    "Trust Store.");
+            return false;
+        }
+        return touchConfigFile();
+    }
+
+    @Override
+    public boolean isConfigured() {
+        boolean isTableExists = certificateMgtDAO.isTableExists();
+        boolean isFilePresent = new File(SSL_PROFILE_FILE_PATH).exists();
+        return isFilePresent && isTableExists;
+    }
+
+    @Override
+    public CertificateMetadataDTO getCertificate(String endpoint, int tenantId) {
+        return certificateMgtDAO.getCertificate("", endpoint, tenantId);
+    }
+
+    @Override
+    public List<CertificateMetadataDTO> getCertificates(int tenantId) {
+        return certificateMgtDAO.getCertificates(tenantId);
+    }
+
+    /**
+     * Modify the sslProfiles.xml file after modifying the certificate.
+     *
+     * @return : True if the file modification is success.
+     */
+    private boolean touchConfigFile() {
+        boolean success = false;
+        File file = new File(SSL_PROFILE_FILE_PATH);
```

Comment:
files not closed use java8 syntacx for file operations

Reply:
In apim-6.1.x we could not use java8 syntax.